File: alexandria/ui-elements/src/displays/components/BlockSelectorContainer.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractBlockSelectorContainer from "../../../gen/displays/components/AbstractBlockSelectorContainer";
import BlockSelectorContainerNotifier from "../../../gen/displays/notifiers/BlockSelectorContainerNotifier";
import BlockSelectorContainerRequester from "../../../gen/displays/requesters/BlockSelectorContainerRequester";
import Block from "./Block";

const styles = theme => ({
	hidden : {
		display: "none"
	}
});

class BlockSelectorContainer extends AbstractBlockSelectorContainer {
	state = {
		selected : 0
	};

	constructor(props) {
		super(props);
		this.notifier = new BlockSelectorContainerNotifier(this);
		this.requester = new BlockSelectorContainerRequester(this);
	};

	render() {
        const { classes } = this.props;
		return (
			<Block styleName={this.props.styleName}
				   layout={this.props.layout}
				   width={this.props.width}>
				{React.Children.map(this.props.children, (child, i) => {
					return (<div className={i !== this.state.selected ? classes.hidden : undefined}>{child}</div>);
				})}
			</Block>
		);
	};

	refreshSelected = (selected) => {
		this.setState({ selected });
	};
}

export default withStyles(styles, { withTheme: true })(BlockSelectorContainer);

File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/BlockSelectorContainer.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.Component;
import io.intino.alexandria.ui.displays.components.selector.Selector;
import io.intino.alexandria.ui.displays.components.selector.SelectorContainer;

public class BlockSelectorContainer<B extends Box> extends AbstractBlockSelectorContainer<B> implements SelectorContainer {
    private int selected;

    public BlockSelectorContainer(B box) {
        super(box);
    }

    @Override
    public void add(Component container) {
        super.add(container);
    }

    @Override
    public void bindTo(Selector selector) {
        selector.onSelect(e -> refreshSelected(e.position()));
    }

    private Component component(String name) {
        return this.children(Component.class).stream().filter(p -> p.name().equalsIgnoreCase(name)).findFirst().orElse(null);
    }

    private void refreshSelected(int position) {
        Component component = child(position);
        if (selected == position) return;
        if (component == null) return;
        this.selected = position;
        notifier.refreshSelected(position);
    }

}